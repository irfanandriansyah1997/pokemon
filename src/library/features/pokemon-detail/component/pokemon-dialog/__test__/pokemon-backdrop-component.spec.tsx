import { mount } from 'enzyme';

import PokemonBackdrop from '@/library/features/pokemon-detail/component/pokemon-dialog/section/pokemon-backdrop.component';

describe(`Testing Pokemon Backdrop Component`, () => {
  describe(`Testing Snapshot Test`, () => {
    it(`Testing With Set Background Image`, () => {
      const ui = mount(
        <PokemonBackdrop
          backgroundImage="image-sample.png"
          show
          zIndex={99999}
        />
      );

      expect(ui.html()).toMatchSnapshot();
    });

    it(`Testing With Set Color`, () => {
      const ui = mount(<PokemonBackdrop color="#ddd" show zIndex={99999} />);

      expect(ui.html()).toMatchSnapshot();
    });

    it(`Testing With Unset Color`, () => {
      const ui = mount(<PokemonBackdrop show zIndex={99999} />);

      expect(ui.html()).toMatchSnapshot();
    });
  });
});
