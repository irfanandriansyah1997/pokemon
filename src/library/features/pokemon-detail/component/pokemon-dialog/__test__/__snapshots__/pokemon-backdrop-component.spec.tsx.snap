// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pokemon Backdrop Component Testing Snapshot Test Testing With Set Background Image 1`] = `"<div class=\\"css-12v3m70\\"></div><div class=\\"css-12v3m70\\"></div>"`;

exports[`Testing Pokemon Backdrop Component Testing Snapshot Test Testing With Set Color 1`] = `"<div color=\\"#ddd\\" class=\\"css-1jaz6a\\"></div><div class=\\"css-g8zdjg\\"></div>"`;

exports[`Testing Pokemon Backdrop Component Testing Snapshot Test Testing With Unset Color 1`] = `"<div class=\\"css-g8zdjg\\"></div><div class=\\"css-g8zdjg\\"></div>"`;
